src/simulation/m_time_steppers.fpp

```diff
@@ -1014,6 +1030,95 @@ contains
 
     end subroutine s_apply_bodyforces
 
+    subroutine s_comprehensive_debug(q_cons_vf, q_prim_vf, t_step, stage)
```

Comment:
This should go in a different module. Maybe a helper file.

Reply:
Addressed